File: livingdoc-fixture-api/src/main/java/org/livingdoc/fixture/api/converter/time/LocalDateConverter.java
package org.livingdoc.fixture.api.converter.time;

import java.time.LocalDate;
import java.time.format.DateTimeFormatter;


public class LocalDateConverter extends AbstractTemporalConverter<LocalDate> {

    @Override
    DateTimeFormatter getDefaultFormatter() {
        return DateTimeFormatter.ISO_DATE;
    }

    @Override
    LocalDate parse(String value, DateTimeFormatter formatter) {
        return LocalDate.parse(value, formatter);
    }

}

File: livingdoc-converters/src/main/kotlin/org/livingdoc/converters/time/LocalDateConverter.kt
package org.livingdoc.converters.time

import java.time.LocalDate
import java.time.format.DateTimeFormatter

/**
 * This converter parses a String to the local date format
 */
open class LocalDateConverter : AbstractTemporalConverter<LocalDate>() {

    override fun defaultFormatter(): DateTimeFormatter = DateTimeFormatter.ISO_LOCAL_DATE
    override fun doParse(value: String, formatter: DateTimeFormatter): LocalDate = LocalDate.parse(value, formatter)

    override fun canConvertTo(targetType: Class<*>) = LocalDate::class.java == targetType
}
